integrated_channels/api/v1/views.py

```diff
@@ -78,3 +83,45 @@ def list(self, request, *arg, **kwargs):
         response = blackboard_serializer.data + canvas_serializer.data + cornerstone_serializer.data + \
             degreed_serializer.data + degreed2_serializer.data + moodle_serializer.data + sap_serializer.data
         return Response(response)
+
+
+class IntegratedChannelHealthCheckView(generics.ListAPIView):
+    """
+        **Use Cases**
+
+            Perform Health Check on user's Canvas connection
+
+        **Example Requests**
+
+            GET /config/health-check
+
+        **Query Parameters for GET**
+
+            * channel_code: The channel code of the CanvasEnterpriseCustomerConfiguration
+            * uuid: The uuid of the CanvasEnterpriseCustomerConfiguration
+
+        **Response Values**
+
+            * is_healthy: Whether the Canvas connection is healthy
+            * health_status: Status code for health status
+
+
+    """
+    renderer_classes = [JSONRenderer, ]
+
+    @method_decorator(require_at_least_one_query_parameter('channel_code'))
+    @method_decorator(require_at_least_one_query_parameter('uuid'))
+    def get(self, request, *args, **kwargs):
+        channel_code = self.request.query_params.get('channel_code')
+        uuid = self.request.query_params.get('uuid')
+
+        # Get enterprise configuration
+        config_type = EnterpriseCustomerPluginConfiguration.get_class_by_channel_code(channel_code)
+        config = config_type.objects.get(uuid=uuid)
```

Comment:
some error handling around channel and config do not exist would be nice here

Reply:
if we're going to release this as is then also probably a list of supported channels with the validation?